packages/client/src/utils/helpers/handleDates.test.ts

```diff
@@ -0,0 +1,235 @@
+          periodEndDate: new Date(''),
+          groupBy: GroupBy.week,
+        }
+      })
+
+      const footprint = { data, error, loading: false }
+      const { missingDates } = checkFootprintDates({ footprint, groupBy })
+
+      expect(missingDates.length > 0).toBe(true)
+    })
+
+    it('has last 30 day total', () => {
+      const dates = [...new Array(4)].map((i, n) =>
+        moment.utc().startOf(groupBy).subtract(n, `${groupBy}s`).toDate(),
+      )
+
+      const data = dates.map((date) => {
+        return {
+          timestamp: date,
+          serviceEstimates: [],
+          periodStartDate: new Date(''),
+          periodEndDate: new Date(''),
+          groupBy: groupBy,
+        }
+      })
+
+      const footprint = { data, error, loading: false }
+      const { missingDates } = checkFootprintDates({ footprint, groupBy })
+
+      expect(missingDates.length > 0).toBe(false)
+    })
+  })
+
+  describe('groupBy month', () => {
+    const groupBy = 'month' as GroupBy.month
+    it('does not have last 30 day total', () => {
+      const data: EstimationResult[] = [
+        {
+          timestamp: moment.utc('2020-01-01').startOf(groupBy).toDate(),
+          serviceEstimates: [],
+          periodStartDate: new Date(''),
+          periodEndDate: new Date(''),
+          groupBy: groupBy,
+        },
+      ]
+
+      const footprint = { data, error, loading: false }
+      const { missingDates } = checkFootprintDates({ footprint, groupBy })
+
+      expect(missingDates.length > 0).toBe(true)
+    })
+
+    it('has last 30 day total', () => {
+      const dates = [...new Array(1)].map((i, n) =>
+        moment.utc().startOf(groupBy).subtract(n, `${groupBy}s`).toDate(),
+      )
+
+      const data = dates.map((date) => {
+        return {
+          timestamp: date,
+          serviceEstimates: [],
+          periodStartDate: new Date(''),
+          periodEndDate: new Date(''),
+          groupBy: groupBy,
+        }
+      })
+
+      const footprint = { data, error, loading: false }
+      const { missingDates } = checkFootprintDates({ footprint, groupBy })
+
+      expect(missingDates.length > 0).toBe(false)
+    })
+  })
+
+  describe('groupBy quarter', () => {
+    const groupBy = 'quarter' as GroupBy.quarter
+    it('does not have last 30 day total', () => {
+      const data: EstimationResult[] = [
+        {
+          timestamp: moment.utc('2020-01-01').startOf(groupBy).toDate(),
+          serviceEstimates: [],
+          periodStartDate: new Date(''),
+          periodEndDate: new Date(''),
+          groupBy: groupBy,
+        },
+      ]
+
+      const footprint = { data, error, loading: false }
+      const { missingDates } = checkFootprintDates({ footprint, groupBy })
+
+      expect(missingDates.length > 0).toBe(true)
+    })
+
+    it('has last 30 day total', () => {
+      const dates = [...new Array(1)].map((i, n) =>
+        moment.utc().startOf(groupBy).subtract(n, `${groupBy}s`).toDate(),
+      )
```

Comment:
Should this be a short helper function at the top of the file? Seems like it's being used often